Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.cpp

```diff
@@ -192,15 +229,106 @@ void NetworkMDNSRegister::registerMDNSName(WebCore::ScriptExecutionContextIdenti
 
 #else // ENABLE_MDNS
 
-void NetworkMDNSRegister::unregisterMDNSNames(WebCore::ScriptExecutionContextIdentifier)
+void NetworkMDNSRegister::unregisterMDNSNames(WebCore::ScriptExecutionContextIdentifier documentIdentifier)
 {
+    for (auto& name : m_perDocumentRegisteredNames.take(documentIdentifier))
+        m_registeredNames.remove(name);
 }
 
 void NetworkMDNSRegister::registerMDNSName(WebCore::ScriptExecutionContextIdentifier documentIdentifier, const String& ipAddress, CompletionHandler<void(const String&, std::optional<WebCore::MDNSRegisterError>)>&& completionHandler)
 {
-    MDNS_RELEASE_LOG("registerMDNSName not implemented");
     auto name = makeString(WTF::UUID::createVersion4(), ".local"_s);
 
+#if USE(GLIB)
+    if (ipAddress == "0.0.0.0"_s || ipAddress == "::"_s) {
+        completionHandler(name, WebCore::MDNSRegisterError::BadParameter);
+        return;
+    }
+
+    if (!m_dbusProxy) {
+        completionHandler(name, WebCore::MDNSRegisterError::Internal);
+        return;
+    }
+
+    m_registeredNames.add(name);
+    m_perDocumentRegisteredNames.ensure(documentIdentifier, [] {
+        return Vector<String>();
+    }).iterator->value.append(name);
+
+    Ref connection = m_connection.get();
+    auto request = makeUnique<PendingRegistrationRequest>(connection.get(), WTFMove(name), ipAddress, sessionID(), WTFMove(completionHandler));
+
+    auto identifier = PendingRegistrationRequestIdentifier::generate();
+    auto addResult = pendingRegistrationCancellablesMap().add(identifier, adoptGRef(g_cancellable_new()));
+    request->identifier = addResult.iterator->key;
+    request->cancellable = addResult.iterator->value;
```

Comment:
For example, you could do:

```suggestion
    request->cancellable = m_cancellable;
```